core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -811,7 +811,16 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
       }
     }
 
+    val isAnonFunc = ".*\\$\\$anonfun\\$\\d+$".r
+    def getNonFuncClass(clz: Class[_]): Option[Class[_]] = {
+      clz.getName match {
+        case isAnonFunc() => getNonFuncClass(clz.getEnclosingClass)
+        case _ => Some(clz)
+      }
+    }
+
```

Comment:
I don't love regex, but at the same time, there's reflection all over this function so I doubt it really matters performance-wise. We could also put the 2.11 specific code in `src/main/scala-2.11` and implement the same method in `src/main/scala-2.12.` but just have it return `Some(clz)`. I don't know that that's necessary though.